cms/djangoapps/api/v1/serializers/course_runs.py

```diff
@@ -29,8 +29,6 @@ class Meta:
 class CourseRunScheduleSerializer(serializers.Serializer):
     start = serializers.DateTimeField()
     end = serializers.DateTimeField()
-    enrollment_start = serializers.DateTimeField(allow_null=True)
```

Comment:
Instead of removing these two fields, why not try with setting `required=False` option? See http://www.django-rest-framework.org/api-guide/fields/#required.
This way, you can test the behavior better to accommodate both the case where enrollment_start and end are supplied, and the case it wasn't supplied.

Reply:
I am able to reproduce this required fields error after modifying the [test_data](https://github.com/edx/edx-platform/blob/ab733f3a1ad81bae1cf4aac617c2e67939638c7a/cms/djangoapps/api/v1/tests/test_serializers/test_course_runs.py#L17-L24) unit test. 

Here is the modification in the test:
```javascript
course_run_data = {
            'id': str(course.id),
            'title': course.display_name,
            'schedule': {
                'start': serialize_datetime(start),
                'end': serialize_datetime(end),
            },
            'team': [
                {
                    'user': instructor.username,
                    'role': 'instructor',
                },
                {
                    'user': staff.username,
                    'role': 'staff',
                },
            ],
            'pacing_type': expected_pacing_type,
        }
        serializer = CourseRunSerializer(data=course_run_data)
        serializer.is_valid(raise_exception=True)
```

**Error:**

```javascript
cms/djangoapps/api/v1/tests/test_serializers/test_course_runs.py:70: in test_data
    serializer.is_valid(raise_exception=True)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = CourseRunSerializer(data={'team': [{'role': 'instructor', 'user': u'robot2'}, ...nImageField(default=<class rest_framework.fields.empty>, source='course_image')
raise_exception = True
    
        if self._errors and raise_exception:
>           raise ValidationError(self.errors)
E           ValidationError: {'schedule': {'enrollment_end': [u'This field is required.'], 'enrollment_start': [u'This field is required.']}}
```

This is because course run data does not have `enrollment_start` and `enrollment_end` fields in `schedule` dict and `CourseRunScheduleSerializer` is expecting both fields.

Reply:
@schenedx as we have removed all the references at publisher end so in my opinion we can remove these two fields from the course run schedule serializer too.

Reply:
Here is the sandbox link:
https://discovery-escalation.sandbox.edx.org/publisher/

Reply:
@Rabia23 I disagree that if we removed enrollment_start and enrollment_end from publisher, we should also remove the ingestion of these data from studio. Here are my reasons:
1. You change here is a API breaking change. Taking away the two fields, and deploy that to production would not be backward compatible with the Publisher code in current production environment. 
2. Do you know if there are other system using this API? How can you be sure Publisher is the only user of this API? 
3. If our openedx community uses this API, the change here will break them as well for the next major openedx releas.
Based on the reasons above, I strongly suggest we set the enrollment_start and enrollment_end fields in serializer to be `required=False` so it is backward compatible.

Reply:
@Rabia23 Thank you for updating tests and showing me that those errors are indeed coming from the serializers. This is an example of good communication!

Reply:
@schenedx you are right. I'll set the enrollment_start and enrollment_end fields in serializer to be `required=False`.